test/src/main/java/org/apache/accumulo/test/functional/CompactionIT.java

```diff
@@ -280,6 +283,28 @@ public void testUserCompactionCancellation() throws Exception {
     }
   }
 
+  @Test
+  public void testErrorDuringUserCompaction() throws Exception {
+    final String table1 = this.getUniqueNames(1)[0];
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
+      client.tableOperations().create(table1);
+      try (BatchWriter bw = client.createBatchWriter(table1)) {
+        for (int i = 1; i <= MAX_DATA; i++) {
+          Mutation m = new Mutation(Integer.toString(i));
+          m.put("cf", "cq", new Value());
+          bw.addMutation(m);
+          bw.flush();
+          // flush often to create multiple files to compact
+          client.tableOperations().flush(table1, null, null, true);
+        }
+      }
+      IteratorSetting setting = new IteratorSetting(50, "error", ErrorThrowingIterator.class);
+      setting.addOption(ErrorThrowingIterator.TIMES, "3");
+      client.tableOperations().attachIterator(table1, setting, EnumSet.of(IteratorScope.majc));
+      client.tableOperations().compact(table1, new CompactionConfig().setWait(true));
```

Comment:
Would be good to read the data in the table after.  Thing it would be good to run this test twice with an additional iterator.  Run the test with an additional iterator that filters a subset of data and after the compaction verify it was filtered.   Run the test another time with an additional iterator that filters all data and verify the scan is empty and the tablet has no files.

Reply:
It seems like you want to test some condition other than an error with this suggestion - the case where the compaction succeeds but there is no output? I could write a different test for this.

Reply:
Modified CompactionIT in 8d3072d . I modified this test method to read the data after compaction and confirm the number of files. I added an additional test that completes a compaction with no resulting data and confirms that there is no file in the tablet metadata.